akka-stream-tests/src/test/scala/akka/stream/impl/ActorInterpreterSpec.scala

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.stream.Supervision._
+import akka.stream.testkit.AkkaSpec
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.testkit.StreamTestKit._
+import akka.stream.impl.fusing.ActorInterpreter
+import akka.stream.stage.Stage
+import akka.stream.stage.PushPullStage
+import akka.stream.stage.Context
+import akka.testkit.TestLatch
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class ActorInterpreterSpec extends AkkaSpec {
+  import FlowGraph.Implicits._
+
+  implicit val mat = ActorFlowMaterializer()
+
+  class Setup(ops: List[Stage[_, _]] = List(fusing.Map({ x: Any ⇒ x }, stoppingDecider))) {
+    val up = PublisherProbe[Int]
+    val down = SubscriberProbe[Int]
+    private val props = ActorInterpreter.props(mat.settings, ops).withDispatcher("akka.test.stream-dispatcher")
+    val actor = system.actorOf(props)
+    val processor = ActorProcessorFactory[Int, Int](actor)
+  }
+
+  "An ActorInterpreter" must {
+
+    "pass along early cancellation" in new Setup {
+      processor.subscribe(down)
+      val sub = down.expectSubscription()
+      sub.cancel()
+      up.subscribe(processor)
+      val upsub = up.expectSubscription()
+      upsub.expectCancellation()
+    }
+
+    "heed cancellation signal while large demand is outstanding" in {
+      val latch = TestLatch()
+      val infinite = new PushPullStage[Int, Int] {
+        override def onPush(elem: Int, ctx: Context[Int]) = ???
+        override def onPull(ctx: Context[Int]) = {
+          Await.ready(latch, 5.seconds)
+          ctx.push(42)
+        }
+      }
+      val N = system.settings.config.getInt("akka.stream.materializer.output-burst-limit")
+
+      new Setup(infinite :: Nil) {
+        processor.subscribe(down)
+        val sub = down.expectSubscription()
+        up.subscribe(processor)
+        val upsub = up.expectSubscription()
+        sub.request(100000000)
+        sub.cancel()
+        watch(actor)
+        latch.countDown()
+        for (i ← 1 to N) withClue(s"iteration $i: ") {
+          try down.expectNext(42) catch { case e: Throwable ⇒ fail(e) }
+        }
```

Comment:
here it would be great with a comment saying something like:
// now the cancellation signal is processed and no more elements are pushed